Formula/dtrx.rb

```diff
@@ -0,0 +1,47 @@
+class Dtrx < Formula
+  include Language::Python::Virtualenv
+
+  desc "Intelligent archive extraction"
+  homepage "https://github.com/dtrx-py/dtrx/"
+  url "https://files.pythonhosted.org/packages/25/cb/1ef093d762f4d5963e9e571daec239acc5f4971eb9daeda77b131d7cf39f/dtrx-8.3.1.tar.gz"
+  sha256 "5587258e762074d5395a6824fd7968ca4f4a1dc481f4852fb84d14e7624433fb"
+  license "GPL-3.0-only"
+
+  # Include a few common decompression handlers in addition to the python dep
+  depends_on "p7zip"
+  depends_on "python@3.9"
+  depends_on "xz"
+  uses_from_macos "bzip2"
+  uses_from_macos "unzip"
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  # Test a simple unzip. Sample taken from unzip formula
+  test do
+    (testpath/"test1").write "Hello!"
+    (testpath/"test2").write "Bonjour!"
+    (testpath/"test3").write "Hej!"
+
+    if OS.mac?
+      system "/usr/bin/zip", "test.zip", "test1", "test2", "test3"
+    else
+      system Formula["zip"].bin/"zip", "test.zip", "test1", "test2", "test3"
+    end
```

Comment:
This will require a `uses_from_macos "zip" => :test`
```suggestion
    system "zip", "test.zip", "test1", "test2", "test3"
```